packages/react/src/utilities/hooks/useRadioGroup/useRadioGroup.ts

```diff
@@ -0,0 +1,149 @@
+  name?: string;
+  /** Set read only state of all radios */
+  readOnly?: boolean;
+  /** Set required state of all radios */
+  required?: boolean;
+  /**
+   * Initial value of the group
+   */
+  value?: string;
+  /** Callback when selected radios changes */
+  onChange?: (nextValue: string, prevValue: string) => void;
+};
+
+/**
+ * Get anything that is set on a radio, but
+ * remove anything that comes from the group itself.
+ */
+export type GetRadioProps = Omit<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  | 'prefix'
+  | 'role'
+  | 'type'
+  | 'size'
+  | 'aria-label'
+  | 'aria-labelledby'
+  | 'label'
+  | 'name'
+  | 'checked'
+  | 'value'
+> & {
+  ref?: React.RefObject<HTMLInputElement>;
+  value?: string;
+};
+
+/**
+ * useRadioGroup is used to group multiple <Radio> components
+ * @example
+ * const { getRadioProps, validationMessageProps, value, setValue } = useRadioGroup({
+ *   disabled: false,
+ *   error: 'Validation message text',
+ *   name: 'group-name',
+ *   onChange: (nextValue, prevValue) => {},
+ *   readOnly: false,
+ *   required: true,
+ *   value: '',
+ * });
+ */
+export function useRadioGroup({
+  error,
+  readOnly,
+  required,
+  disabled,
+  name,
+  onChange,
+  value: initalValue = '',
+}: UseRadioGroupProps = {}) {
+  const [groupValue, setGroupValue] = useState(initalValue);
+  const errorId = useId();
+  const namedId = useId();
+  const radioGroupName = name || namedId;
+
+  return {
+    /**
+     * Current value of the group.
+     */
+    value: groupValue,
+    /**
+     * Set the value of the group.
+     *
+     * @param value string[]
+     * @returns void
+     */
+    setValue: setGroupValue,
+    /**
+     * Props to send to the `Radio` component.
+     * @example
+     * <Radio label="Option 1" {...getRadioProps('option-1')} />
+     */
+    getRadioProps: (propsOrValue: string | GetRadioProps) => {
+      const props =
+        typeof propsOrValue === 'string'
+          ? { value: propsOrValue }
+          : propsOrValue;
+      const { ref = undefined, value = '', ...rest } = props;
+      const localRef = useRef<HTMLInputElement>(null);
+      const mergedRefs = useMergeRefs([ref, localRef]);
+
+      const handleChange = () => {
+        const input = localRef.current;
+        const isInput = input instanceof HTMLInputElement;
+
+        if (isInput && input.name === radioGroupName) {
+          setGroupValue((prevValue) => {
+            onChange?.(input.value, prevValue);
+            return input.value;
+          });
+        }
+      };
+
+      useEffect(() => {
+        if (!localRef.current) return;
+        localRef.current.checked = value === groupValue;
+      }, [groupValue, value]);
+
+      return {
+        /* Spread anything the user has set first */
+        ...rest,
+        /* Concat ours with the user prop */
+        name: radioGroupName,
+        'aria-describedby':
+          `${(!!error && errorId) || ''} ${rest['aria-describedby'] || ''}`.trim(),
```

Comment:
Ikke legge til om den ikke har verdi? ☺️ 
```suggestion
        'aria-describedby':
          `${(error ? errorId : ''} ${rest['aria-describedby'] || ''}`.trim() || undefined,
```

Reply:
Good catch! This will also needs to be fixed for `useCheckboxGroup`. I'll fix!

Reply:
We can just pass their `rest['aria-describedby']` though? If it is not set then it is undefined and should not set it in the dom either

Reply:
we could, but we need to merge with `errorId` so that makes it hard to just pass directly? ☺️